app/questionnaire/questionnaire_manager.py

```diff
@@ -22,18 +22,16 @@ class QuestionnaireManager(object):
     This class represents a user journey through a survey. It models the request/response process of the web application
     using a doubly linked list. Each node in the list is essentially a page displayed to the user. A new node is created
     by a GET request and subsequently updated via a POST request.
-
     The doubly linked list approach allows us to maintain the path the user has taken through the question. If that path
     changes we archive off the nodes in case the user revisits that path.
-
     '''
     def __init__(self, schema):
         self.submitted_at = None
-        self._schema = schema
-        self._current = None  # the latest node
-        self._first = None  # the first node in the doubly linked list
-        self._archive = {}  # a dict of discarded nodes for later use (if needed)
-        self._valid_locations = self._build_valid_locations()
+        self.schema = schema
+        self._current = None  # the latest page
+        self._first = None  # the first page in the doubly linked list
+        self.archive = {}  # a dict of discarded pages for later use (if needed)
```

Comment:
Not sure we should be renaming these variables, as this is essentially exposing the internals of the class